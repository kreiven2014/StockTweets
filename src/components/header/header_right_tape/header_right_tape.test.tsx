/* REACT */
import React from "react";
// import { Alert } from "react-native";

/* MODULES */
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

/* CUSTOM MODULES */
import HeaderRightTape from "./header_right_tape";

describe("HeaderRightTape tests", () => {
  let component: any;

  beforeEach(() => {
    component = shallow(<HeaderRightTape />);
  });

  test("renders correctly", (): void => {
    const tree = renderer.create(<HeaderRightTape />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders null when isn't visible", (): void => {
    component.setState({
      visible: false,
    });
    expect(component.state().visible).toEqual(false);

    const tree = renderer.create(<HeaderRightTape />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("HeaderRightTape tests with dev env", () => {
  let component: any;
  let HeaderRightTapeComp: any;

  beforeEach(() => {
    jest.resetModules();
    jest.doMock("src/configs/main_config", () => {
      // $FlowFixMe
      const config = require.requireActual("src/configs/main_config");

      return {
        ...config,
        ENV: "dev",
      };
    });

    // @Note: need to require module again after do mock of configs
    HeaderRightTapeComp = require("./header_right_tape").default;
    component = shallow(<HeaderRightTapeComp />);
  });

  test("renders correctly", (): void => {
    const tree = renderer.create(<HeaderRightTapeComp />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should call the onPress callback", () => {
    expect(component.state().visible).toEqual(true);
    // TODO(ANY): adapter don't return tree of react-native elements.
    // should return TouchableOpacity
    const button = component
      .find({
        testID: "headerRightTape",
      })
      .first();
    button.simulate("longPress");
    expect(component.state().visible).toEqual(false);

    // @Note: left this for example how to test when on press is calling alert
    // button.simulate("press");
    // expect(Alert.alert).toHaveBeenCalled();

    // // TODO(anybody): check that functions are called corresponding function from props !
    // Alert.alert.mock.calls[0][2][0].onPress();
    // Alert.alert.mock.calls[0][2][1].onPress();
  });
});
