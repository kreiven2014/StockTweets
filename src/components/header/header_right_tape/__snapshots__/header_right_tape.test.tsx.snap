// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRightTape tests renders correctly 1`] = `null`;

exports[`HeaderRightTape tests renders null when isn't visible 1`] = `null`;

exports[`HeaderRightTape tests with dev env renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF0000",
      "height": 34.5,
      "marginRight": -60,
      "marginTop": 85.5,
      "opacity": 1,
      "position": "absolute",
      "right": 0,
      "transform": Array [
        Object {
          "rotate": "45deg",
        },
      ],
      "width": 300,
    }
  }
  testID="headerRightTape"
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
          "letterSpacing": 0.6,
          "lineHeight": 27.5,
          "textAlign": "left",
        },
        undefined,
        Object {
          "color": "#FFFFFF",
          "fontSize": 17,
          "fontWeight": "900",
          "lineHeight": 27.5,
        },
      ]
    }
  >
    dev
29.11.19 16-30
  </Text>
</View>
`;
